test/mpi/configure.ac

```diff
@@ -264,35 +263,6 @@ AC_ARG_WITH(dtpools-datatypes,
 # parse args for typegen.sh script
 dtp_args="--with-dtpools-datatypes=${with_dtpools_datatypes}"
 
-AC_ARG_WITH(dtpools-maxbufsize,
-    [AC_HELP_STRING([--with-dtpools-maxbufsize=size_in_bytes],
-        [maximum buffer size needed for the generated datatypes.])],,)
-
-AC_CHECK_SIZEOF(void *)
-if test ! -z "${with_dtpools_maxbufsize}" ; then
-    # if the user specified a size, use it; otherwise try to
-    # autodetect it
-    dtp_args="$dtp_args --with-dtpools-maxbufsize=${with_dtpools_maxbufsize}"
-elif test "$ac_cv_sizeof_void_p" -le "4" ; then
-    # this is somewhat of a hack; the 32-bit FreeBSD system that we
-    # have access to is pretty crappy with just 3GB of memory.  so
-    # reduce the memory size of each DTP object, so we don't run out
-    # of memory
-    if test ! -z "`echo $host_os | grep ^freebsd`" ; then
-       # 32-bit FreeBSD gets 128MB per process
-       dtp_args="$dtp_args --with-dtpools-maxbufsize=134217728"
-    else
-       # remaining 32-bit machines get 512MB per process
-       dtp_args="$dtp_args --with-dtpools-maxbufsize=536870912"
-    fi
-else
-    # all other machines get the default memory per process
-    dtp_args="$dtp_args --with-dtpools-maxbufsize=2147483648"
-fi
-
-export dtp_args
-AC_CONFIG_COMMANDS([gentests],[$srcdir/maint/gentests_dtp.sh $dtp_args],[dtp_args="$dtp_args"])
```

Comment:
I think `AC_CONFIG_COMMANDS` already runs it at the end.  So I don't understand this part of the patch.

Reply:
`AC_CONFIG_COMMANDS` is more complex -- It is the shell code (in `configure`) dumping statement into `config.status` and then `config.status` runs it as if it is one of the `AC_OUTPUT`. In doing so, it become necessary to understand how the variable is being substituted at which stage, hence the `export` and separate argument. I don't expect average developer readily understand these details.

There is use case for `AC_CONFIG_COMMANDS`, but our case is a very straight forward case and I think just call the script directly is much simpler.

Reply:
With your change, you removed the ability for `config.status` to rerun that command.  So, for the cases where `make` has to reconfigure, it'll not regenerate the tests.  By bypassing these macros, you are taking away some of the functionality that they provide.

Reply:
I see. That is a good point.

In this case, the "tag" is a pseudo one, so I am not sure how it helps. When `make` has to reconfigure, it'll rerun the `configure` which will run the `gentests` script again, right?

Reply:
Correct.  I agree that the `tag` is not very useful.